src/buffer_manager.cc

```diff
@@ -105,4 +105,29 @@ void BufferManager::clear_buffer_trash()
     m_buffer_trash.clear();
 }
 
+void BufferManager::arrange_buffers(ConstArrayView<String> first_ones)
+{
+    Vector<size_t> indices;
+    for (const auto& name : first_ones)
+    {
+        auto it = find_if(m_buffers, [&](auto& buf) { return buf->name() == name or buf->display_name() == name; });
+        if (it == m_buffers.end())
+            throw runtime_error{format("no such buffer '{}'", name)};
+        size_t index = it - m_buffers.begin();
```

Comment:
`m_buffers` is a vector, why not just move items at its beginning directly instead of storing indexes?

Reply:
Because at this point we can still throw, so we don't want to manipulate `m_buffers` just yet

Reply:
Fair enough, note that you should use `std::distance` to calculate the amount of items between two iterators.

Reply:
I don't see a good reason why, and this pattern is used in several places in the codebase so I will keep it as is.